File: LeetCode/Easy Level/pivotNumber.py
def pivotIndex(nums):
    S = sum(nums)
    lsum = 0
    for i, x in enumerate(nums):
        if (S - lsum -x) == lsum:
            return i
        else:
            lsum += x
        
    return -1



print(pivotIndex([1,7,3,6,5,6]))
print()
print(pivotIndex([1,2,3]))
print()
print(pivotIndex([2,1,-1]))
print()
print(pivotIndex([-1,-1,-1,-1,-1,-1]))

File: LeetCode/Easy Level/pivotNumber.js
const pivotIndex = (nums)=>{
    let S = 0;
    for(let n of nums){
        S += n;
    }
    let lsum = 0;
    for(let i=0; i<nums.length; i++){
        if(lsum == (S - lsum -nums[i])){
            return i;
        } else {
            lsum += nums[i]
        }
    }

    return -1
}

console.log(pivotIndex([1,7,3,6,5,6]));
